Lib/python/pyrun.swg

```diff
@@ -127,7 +127,12 @@ SWIG_Python_AppendOutput(PyObject* result, PyObject* obj) {
     if (!PyList_Check(result)) {
       PyObject *o2 = result;
       result = PyList_New(1);
-      PyList_SetItem(result, 0, o2);
+      if (result) {
+        PyList_SET_ITEM(result, 0, o2);
+      } else {
+        Py_DECREF(obj);
```

Comment:
The protocol for `SWIG_Python_AppendOutput` is that it always steals a reference to `result` and `obj`, and returns a referenced object. If there is an error, this still holds - but an exception may be raised.

In this further OOM fix, we ensure that a reference to `obj` is always consistently stolen.